event/listener.php

```diff
@@ -221,4 +227,36 @@ public function show_idea($event)
 			'FORUM_NAME'		=> $this->language->lang('IDEAS'),
 		));
 	}
+
+	/**
+	 * Show users as viewing Ideas on Who Is Online page
+	 *
+	 * @param object $event The event object
+	 * @return null
+	 * @access public
+	 */
+	public function viewonline_ideas($event)
+	{
+		if ($event['on_page'][1] == 'app')
+		{
+			if (strrpos($event['row']['session_page'], 'app.' . $this->php_ext . '/ideas/post') === 0)
+			{
+				$event['location'] = $this->language->lang('POSTING_NEW_IDEA');
+				$event['location_url'] = $this->helper->route('phpbb_ideas_index_controller');
+			}
+			else if (strrpos($event['row']['session_page'], 'app.' . $this->php_ext . '/ideas') === 0)
```

Comment:
Not .com uniqueness but any site using their own front controller. Which is
a completely ordinary thing in good practice web development. Also, this is
only realistically going to be used on .com and maybe a few other
developers sites, the kind of people likely to have such changes in place.
There must be an easy way to see if we are in the front controller. I'll
have a think. @Nicofuma.

--
Michael C

On 9 November 2015 at 23:31, Matt Friedman <notifications@github.com> wrote:

> In event/listener.php
> <https://github.com/phpbb/phpbb-ideas/pull/36#discussion_r44349826>:
>
> > +	 * Show users as viewing Ideas on Who Is Online page
> > +	 *
> > +	 * @param object $event The event object
> > +	 * @return null
> > +	 * @access public
> > +	 */
> > +	public function viewonline_ideas($event)
> > +	{
> > +		if ($event['on_page'][1] == 'app')
> > +		{
> > +			if (strrpos($event['row']['session_page'], 'app.' . $this->php_ext . '/ideas/post') === 0)
> > +			{
> > +				$event['location'] = $this->language->lang('POSTING_NEW_IDEA');
> > +				$event['location_url'] = $this->helper->route('phpbb_ideas_index_controller');
> > +			}
> > +			else if (strrpos($event['row']['session_page'], 'app.' . $this->php_ext . '/ideas') === 0)
>
> Well, app.php is an extension's front controller page. Ugly or not, this
> is the way we tell if a user is viewing an extensions front controller, and
> it works for extensions in general. If it doesn't work with dot com's
> uniqueness, that will have to be dealt with separately for dot com.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phpbb/phpbb-ideas/pull/36/files#r44349826>.
>